src/lwc/geReviewDonations/geReviewDonations.js

```diff
@@ -132,30 +168,42 @@ export default class geReviewDonations extends NavigationMixin(LightningElement)
     }
 
     /*******************************************************************************
-    * @description Method receives an event from the child geDonationMatching
-    * component and sets the currently selected donation along with its type.
-    * 
-    * @param {object} event: Custom Event object received from child component.
+    * @description Receives an event from the modal component geDonationMatching,
+    * locally stores and dispatches an event up to parent geFormRenderer of the
+    * currently selected donation along with its type.
+    *
+    * @param {object} pubsubEvent: Pubsub event fired from geDonationMatching
+    * containing the data on the user selected donation.
     */
-    handleReceiveEvent(event) {
-        if (event.detail.hasOwnProperty(PAYMENT)) {
-            this.selectedDonation = event.detail.payment;
-            this.donationType = PAYMENT;
-        } else if (event.detail.hasOwnProperty(OPPORTUNITY)) {
-            this.selectedDonation = event.detail.opportunity;
-            this.donationType = OPPORTUNITY;
-        } else {
-            this.selectedDonation = this.donationType = undefined;
-        }
+    handleReceiveEvent(pubsubEvent) {
+        this.setSelectedDonationsData(pubsubEvent);
 
         const detail = {
-            selectedDonation: deepClone(this.selectedDonation),
-            donationType: deepClone(this.donationType)
+            selectedDonation: deepClone(this._selectedDonation),
+            donationType: deepClone(this._donationType)
         }
-
         this.dispatchEvent(new CustomEvent('changeselecteddonation', { detail }));
     }
 
+    /*******************************************************************************
+    * @description Sets local properties related to the currently selected donation.
+    * 
+    * @param {object} pubsubEvent: Event object sent from the pubsubNoPageRef component.
+    */
+    setSelectedDonationsData(pubsubEvent) {
```

Comment:
Is pubsub needed here?  Is this used when the Renderer's selected donation changes?

Reply:
Yeah, I'd argue that this is needed. The pubsub event fired by the web component (`geDonationMatching`) that's contained within the modal that's created via the overlay library. So the flow of data is basically as follows once the modal is open and the user has made a donation selection...

- `geDonationMatching` fires a pubsub event to notify `geReviewDonations` of the selection.
- `geReviewDonations` updates the messaging/links for the UI and dispatches a standard custom event to `geFormRenderer`.
- `geFormRenderer` receives the standard custom event and sets property values for the various donation imported fields that's used to build the data import record on save.

Reply:
Ok - sounds like we have a flow that is working well.  Thanks for the detail!